src/transports/tcp/TcpConnectionFactory.cpp

```diff
@@ -72,27 +71,11 @@ TcpConnectionFactory::~TcpConnectionFactory() {
   VLOG(1) << "Destroying TcpConnectionFactory";
 }
 
-void TcpConnectionFactory::connect(OnConnect cb) {
+void TcpConnectionFactory::connect(OnDuplexConnectionConnect cb) {
   worker_.getEventBase()->runInEventBaseThread(
       [ this, fn = std::move(cb) ]() mutable {
         new ConnectCallback(address_, std::move(fn));
       });
 }
 
-std::unique_ptr<DuplexConnection>
```

Comment:
please keep this method as it is used by the RT GW.

the body of the method should be just:

return std::make_unique<TcpDuplexConnection>(
      std::move(socket), eventBase, std::move(stats));